File: src/android/app/src/main/java/org/citra/citra_emu/features/settings/model/view/DateTimeSetting.java
package org.citra.citra_emu.features.settings.model.view;

import org.citra.citra_emu.features.settings.model.Setting;
import org.citra.citra_emu.features.settings.model.StringSetting;

public final class DateTimeSetting extends SettingsItem {
    private String mDefaultValue;

    public DateTimeSetting(String key, String section, int titleId, int descriptionId,
                           String defaultValue, Setting setting) {
        super(key, section, setting, titleId, descriptionId);
        mDefaultValue = defaultValue;
    }

    public String getValue() {
        if (getSetting() != null) {
            StringSetting setting = (StringSetting) getSetting();
            return setting.getValue();
        } else {
            return mDefaultValue;
        }
    }

    public StringSetting setSelectedValue(String datetime) {
        if (getSetting() == null) {
            StringSetting setting = new StringSetting(getKey(), getSection(), datetime);
            setSetting(setting);
            return setting;
        } else {
            StringSetting setting = (StringSetting) getSetting();
            setting.setValue(datetime);
            return null;
        }
    }

    @Override
    public int getType() {
        return TYPE_DATETIME_SETTING;
    }
}

File: src/android/app/src/main/java/org/citra/citra_emu/features/settings/model/view/DateTimeSetting.kt
// Copyright 2023 Citra Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.

package org.citra.citra_emu.features.settings.model.view

import org.citra.citra_emu.features.settings.model.AbstractSetting
import org.citra.citra_emu.features.settings.model.AbstractStringSetting

class DateTimeSetting(
    setting: AbstractSetting?,
    titleId: Int,
    descriptionId: Int,
    val key: String? = null,
    private val defaultValue: String? = null
) : SettingsItem(setting, titleId, descriptionId) {
    override val type = TYPE_DATETIME_SETTING

    val value: String
        get() = if (setting != null) {
            val setting = setting as AbstractStringSetting
            setting.string
        } else {
            defaultValue!!
        }

    fun setSelectedValue(datetime: String): AbstractStringSetting {
        val stringSetting = setting as AbstractStringSetting
        stringSetting.string = datetime
        return stringSetting
    }
}
